### PR TITLE
Implement FE/profile details page

### PR DESCRIPTION
**Pull request by esteban-gs**
#31 
### What this PR does (required):
- Each Profile will have its own dedicated page that outlines their profile in detail
- Profile id gets passed from query params and fetched on `useEffect`
- Api call for profiles
- Sample data to showcase profile page
- Sample review to showcase reviews list
- Book a sitter from their profile page

### Screenshots / Videos (front-end only):
image

### Any information needed to test this feature (required):
- Log into front-end
- Go to `/profile/anything` <- temp demo route

### Any issues with the current functionality (optional):
- Needs integration with back end

### CODE REVIEW

File: client/src/pages/Profile/ProfileDetail.tsx/ProfileDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const { id } = useParams<ProfileDetailParams>();
+  const [demoProfile, setDemoProfile] = useState<IDemoProfile>(demoProfiles[0]); // static until backend inplemented
+  const [_demoReviews, setDemoReviews] = useState<IDemoReview[]>(demoReviews); // static until backend inplemented
+
+  useEffect(() => {
+    // TODO call when API exists
+    // getProfile(id).then((res) => {
+    //   setDemoProfile(res as ProfileDetails);
+    // });
+  }, [id]);
```
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileDetail.tsx/ProfileDetail.tsx:23]: If code is not used please remove it to keep your code clean.

File: client/src/pages/Profile/ProfileDetail.tsx/ProfileDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          </Grid>
+          <Grid item xs={12}>
+            <Divider variant="middle" sx={{ marginTop: '20px', marginBottom: '20px' }}>
+              Reviews
+            </Divider>
+          </Grid>
+          <Grid item xs={12} container>
+            <Grid item xs container direction="column" spacing={2}>
+              {_demoReviews.map((item: IDemoReview, index: number) => (
+                <ProfileReview key={index} review={{ ...item }} />
```
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileDetail.tsx/ProfileDetail.tsx:71]: Using index as a key is an anti-pattern in React and is discouraged. You can use another field for you key.

File: client/src/pages/Profile/temp/demo_data.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+export interface IDemoProfile {
+  id: string;
+  name: string;
+  description: string;
+  photo: string;
+}
+
+// placeholder
```
Comment (ezeilo-su) [inline client/src/pages/Profile/temp/demo_data.ts:8]: Remove comment

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
UI looks good! Kindly address those comment that I left here.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Could we add in the second column with the booking form too? 🙂

**Review by moffatethan — APPROVED** [review]
Great job!!

**Pull request merged**